src/components/pop-up/index.tsx

```diff
@@ -0,0 +1,57 @@
+import styled from '@emotion/styled'
+import React from 'react'
+
+type Props = {
+  children: React.ReactNode
+  placeInView: [
+    isDisplayed: boolean,
+    setDisplay: React.Dispatch<React.SetStateAction<boolean>>,
+  ]
+}
+
+function PopUp({ children, placeInView }: Props) {
+  const [isDisplayed, setDisplay] = placeInView
+  const clickListenerRef = React.useRef<HTMLDivElement | null>(null)
+
+  React.useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        clickListenerRef.current &&
+        clickListenerRef.current !== event.target
```

Comment:
this is a bug here it should be` !clickListenerRef.current.contains(event.target)` but couldn't put Type on target

Reply:
MouseEvent receives an HTML element type as a generic, as seen [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f97b85aa538e6f844efd3884af73baae54eb5ae6/types/react/index.d.ts#L1234), also prefer `currentTarget` over `target` for the reasons explained [here](https://stackoverflow.com/questions/10086427/what-is-the-exact-difference-between-currenttarget-property-and-target-property)

Reply:
Ohhhh I see the problem, sorry I misunderstood lol

Reply:
np ^_^